### DISCUSSION TITLE
Error validating datasource `db` : the URL must start with the protocol `postgresql://` or `postgres://`

### DISCUSSION THREAD

**Discussion by mabdullahadeel** [prisma/prisma#12170] [Q&A]
I am trying to run simple migration using `prisma migrate dev`. I have my `.env` file set up in the root directory of my application.

```.env
DATABASE_URL="postgresql://<postgres>:<postgres>@localhost:5432/<databasename>?schema=public"
```

I have defined the `DATABASE_URL` in my `.env` file. But when I try to run migrations, I am getting the following logs.

```shell
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Error: Get config: Schema Parsing P1012

error: Error validating datasource `db`: the URL must start with the protocol `postgresql://` or `postgres://`.
  -->  schema.prisma:10
   | 
 9 |   provider = "postgresql"
10 |   url      = env("DATABASE_URL")
   | 

Validation Error Count: 1
```

I have tried:

 - Downgrading the versions to `@prisma/cli@3.8.1` `prisma@3.8.1`
 - moving `.env` file to `./prisma/.env`

**Comment by isNan909** [comment]
I also faced the same issue. I rechecked the connection string generated from free Postgres instance of Heroku, did not help.

I rechecked on changing it to local DB nothing helped. I do have repo of this. 

But I really don't know how to reproduce this issue.

**Reply by mabdullahadeel** [reply]
Yeah for me, this error does not appear if I directly paste the connection string in my `schema.prisma` file. Apparently, there is something wrong going on with loading `.env` file variables. I have even tried moving to `./prisma/.env`. Running migrations seems to pick up the `.env` file just fine as per the logs.

> Environment variables loaded from .env

**Reply by isNan909** [reply]
Directly pasting the connection string inside schema file didn't work for me either. 

But thanks for reply.

**Reply by mabdullahadeel** [reply]
can you please share your connection string?

**Reply by isNan909** [reply]
It's a Heroku Postgres instance free hobby plan. I  think it's unsafe to  share connection sting in public.

Please connect with me in Prisma official slack 😁

**Reply by mabdullahadeel** [reply]
No, you can just remove the username, password, and even database name from the connection string. Just replace those with some dummy values. I have done the same here 👇.

> DATABASE_URL="postgresql://postgres:postgres@localhost:5432/databasename?schema=public"

**Reply by mabdullahadeel** [reply]
Just make sure that you are using `postgresql` as protocol. I have found some solutions where this resolved the issue.

**Reply by isNan909** [reply]
Connection string

DATABASE_URL="postgresql://oxdyavvvddzceh:XXX@ec2-3-227-195-74.compute-1.amazonaws.com:5432/dcb8cd0lc5j1lp?"

**Reply by mabdullahadeel** [reply]
have you tried this

```sh
DATABASE_URL="postgresql://oxdyavvvddzceh:XXX@ec2-3-227-195-74.compute-1.amazonaws.com:5432/dcb8cd0lc5j1lp?schema=public"
```
try adding `schema=public` at the end of the connection string.
try pasting this directly into your `schema.prisma` file.

**Reply by mabdullahadeel** [reply]
Also, I have noticed one more thing. For me, this error seems to appear just with `Postgres`. Aso soon as I switched to `mysql` on the local instance. Everything worked just fine.

> NOTE: I am not directly pasting `connection string` . It seems to pick it from `.env` file just fine.

**Reply by idirmosh** [reply]
I just run to this issue when switching my project to a monocrop, in my case I'm using turborepo and I have initialized Prisma in the Packages directory I also have my database connection there in a separate folder cause i wanted to share my apps so i was doing most of the operation from `apps/api/graphql`.  but this did seem to work cause Prisma doesn't know how to figure out the db url located in `packages/prisma/.env`

whenever i try to query my graphql api i get this error on the console
```ts
api:dev: error - unhandledRejection: Error: error: Error validating datasource `db`: the URL must start with the protocol `mysql://`.
api:dev:   -->  schema.prisma:8
api:dev:    | 
api:dev:  7 |   provider             = "mysql"
api:dev:  8 |   url                  = env("DATABASE_URL")
api:dev:    | 
api:dev: 
api:dev: Validation Error Count: 1
api:dev: (node:18112) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 1)
```
after hours of debugging i found a workaround by programmatically override a datasource url as shown bellow

```ts
const prisma = new PrismaClient({ datasources: {  db: { url: "mysql://yourdburlstringhere" } } });
```

so for anyone who has this issue in the future this is how i worked around the error.

**Reply by janpio** [reply]
Can you please turn this into a standalone issue @idirmosh and provide all the information (and optimally a simple reproduction project) we can use to reproduce this? Thanks.

**Reply by idirmosh** [reply]
I'll be on it ASAP.

**Reply by idirmosh** [reply]
@janpio I have created a new monorepo to reproduce the issue I mentioned above following the same setup as for now everything works perfectly  and the datasource error is no where to be seen, so I probably missed up something on my main monorepo that's why Prisma wasn't happy, I will dig into this more and I will keep you posted if the issue is still present.

**Reply by KamilJakobczak** [reply]
> 
> after hours of debugging i found a workaround by programmatically override a datasource url as shown bellow
> 
> ```ts
> const prisma = new PrismaClient({ datasources: {  db: { url: "mysql://yourdburlstringhere" } } });
> ```
> 
> so for anyone who has this issue in the future this is how i worked around the error.

I just ran into the same problem even though it was working perfectly fine with environment variable before. This is the only thing that solved it Thanks!

**Reply by DavidEGonsalves** [reply]
I'm running into this exact error.  Can you tell me where you placed your workaround?  In schema.prisma?

Thanks!

**Reply by Anupkp19** [reply]
this worked
Thanks

**Reply by jarrodmedrano** [reply]
This won't work for me because I need to generate.

**Reply by meAasifAli** [reply]
i did this but server is throwing me another error the URL must start with the protocol `prisma://` then when i added this then it is throwing me another error url must have a valid api key

**Reply by rakshittt** [reply]
can you tell me , where to add this line of code. I am facing the same error

**Reply by jpenna** [reply]
The code suggested by @KamilJakobczak is used to create the Prisma client. You use that in the JS file where you create the Prisma client. It isn't intended for prisma files.

**Comment by juanitoddd** [comment]
I was having the same issue, having prisma inside a Docker Container.
I realize I had set env variable `DATABASE_URL` inside my docker-compose.yml so, the .env was being override. 
I know is kind of stupid, but maybe your `DATABASE_URL` env variable is being override without you noticing it.

**Reply by mabdullahadeel** [reply]
But the thing in my case was if I set `DATABASE_URL` to a `mysql` db, everything worked. I was just facing the problem with `postgres`

**Comment by hkrhasan** [comment]
![image](https://user-images.githubusercontent.com/48556585/183467084-5090a12f-0713-4cc1-9737-fbac427409f9.png)

I also face the same issue 😞

**Reply by tchaffee** [reply]
I am seeing this even with an `sqlite` provider. In fact, I got the error following the quickstart tutorial, which doesn't inspire confidence.

**Reply by Easy-Cloud-in** [reply]
remove the "" in .env file
DATABASE_URL=postgresql://postgres:postgres@host.docker.internal:49153/testdb
it worked for me,  but i get  P1001: Can't reach database server at `host.docker.internal`:`49153` error.

**Reply by james-gardner** [reply]
oh I feel so so foolish.

**Reply by zeke** [reply]
Removing the quotes worked for me as well. Most .env parsers support quotes around values, but apparently not prisma.

**Reply by zisra** [reply]
Still doesn't fix it for me

**Reply by MuksithMRA** [reply]
Thanks bro it worked

**Reply by ajfahim** [reply]
this worked for me.

**Reply by ayoubshah2720** [reply]
This also worked for me.

**Reply by subrotoice** [reply]
DATABASE_URL = "mysql://root:@localhost:3306/issue-tracker";   - Not Working
DATABASE_URL = "mysql://root:@localhost:3306/issue-tracker"    - Working

**Reply by Phinicz** [reply]
> remove the "" in .env file DATABASE_URL=postgresql://postgres:[postgres@host.docker.internal](mailto:postgres@host.docker.internal):49153/testdb it worked for me, but i get P1001: Can't reach database server at `host.docker.internal`:`49153` error.

I love you

**Reply by wrspada02** [reply]
> remove the "" in .env file DATABASE_URL=postgresql://postgres:[postgres@host.docker.internal](mailto:postgres@host.docker.internal):49153/testdb it worked for me, but i get P1001: Can't reach database server at `host.docker.internal`:`49153` error.

Thank you. Looks good now.

**Reply by musamaashraf** [reply]
Removing quotes from the .env file worked for me as well.

**Comment by rhinodavid** [comment]
I had this problem while working with the next-auth demo repo. The problem ended up being a _second_ declaration of `DATABASE_URL` at the very bottom of the `.env` file.

**Reply by seanlowe** [reply]
My issue was, like this, being caused by a second declaration of `DATABASE_URL`.. except mine was in my `.env.example` file.

**Comment by joeydotdev** [comment]
Ran into this similar error but only in the context of a production box and not local. 

My issue was that my environment variable was in the format of `DATABASE_URL='connectionstring'` instead of `DATABASE_URL=connectionstring`. After I changed to the latter, connections were succeeding.

**Reply by Fritiof** [reply]
This solved it for me! THANKS. Can't believe that's a problem tbh

**Reply by lorenzocesconetto** [reply]
THANK YOU! Omg this bug is so annoying

**Reply by sorenowy** [reply]
This is gold. Many thanks!
I had stuck multiple times on providing data via `' '` - but i simply didn't noticed that in docs.

**Reply by maag070208** [reply]
THANK YOU!!!!!!!!!

**Reply by teetangh** [reply]
This solved it for me as well. Sometimes(like in my case) special characters in username/password could also be posing a threat. So try to keep things simple and get project working. Once the prototype is working, you can make the creds harder.

**Comment by rolki-png** [comment]
If you recently swapped your DATABASE_URL, make sure you run `npx prisma generate` then try it again

**Reply by d3r3nic** [reply]
I simply re-ran my pipeline and since the pipeline runs npx prisma generate when building, it  fixed my issue. 
very odd error! a very simple solution that no one would think about!
thank you

**Reply by theborowski** [reply]
Thanks! This helped me out as well. At first I thought it was also related to the answers above with using quotes but I just had to regenerate.

**Reply by safwansamsudeen** [reply]
Insane, man. Thanks so much!

**Reply by RobsonTorres** [reply]
Obrigato. Isso resolveu o meu problema, ótima dica.

**Comment by oota-bear** [comment]
※I'm sorry if it's a strange English because I'm translating it into English with google translate.

I had the same error, but I resolved it, so I'll post it here.

①There are two prisma.schema files, and the prisma described was wrong
→Project created in ”furourio”.
because 「server/prisma/prisma.schema」「prisma/prisma.shema」two prisma.schema file exist.
i was fixing 「server/prisma/prisma.schema」 but 「prisma/prisma.shema」

Also, prisma/prisma.shema」 did not have 「.env」, so it was created and migrated successfully

**Reply by DavidEGonsalves** [reply]
This is very similar to what happened to me.  I'd created my project in the same folder as a school project I was using as an example.  Prisma's smart enough to search up through the directory tree & find the .env file from the school project (which was using PostGresSQL & not MySQL).  My clue came in the error message when it said that it was reading two .env files, one in the local \prisma\ directory, and another in the ../ directory.  That's when I figured out that the actual error was two conflicting .env files.  The Prisma error message could have been better, it just kept telling me that "mysql" had to be the database string, not that there were two database strings it was trying to resolve.  Anyway, I've figured out my problem, and hopefully this (and your solution) will help someone down the road.

 Thanks for the reply, I appreciate it, and Google translate did a pretty good job.

**Reply by oota-bear** [reply]
Thanks for your reply.
It's hard to understand which .env is related to which...
I'm happy to be of some help！

**Comment by michaelwell23** [comment]
Importing the 'dotenv-safe/config inside the jest.config file is what worked for me.

`require('dotenv-safe/config')
const { exec } = require('child_process')

process.env.DB_URL = `${process.env.DB_URL}_testdb?schema=test_schema`

exec('yarn db:migrate')
`

**Comment by theodore-zd** [comment]
I had this exact issue today, in a **nextjs** project using **next-auth**  and **postgres** as my db after double checking it wasn't something silly like a spelling error, removed my .next folder and all contents, ran `next dev` and it starting working no problem.

Hope this helps 👍🏻

**Comment by ELORCHI** [comment]
I had the same problem today. what i found is that prisma tried to read the env variable from the os exported env variables, because i added the .env to my docker-compose.yml. the solution i found is unseting the env DATABSE_URL after an exec into the container.
the weird thing here is that the two env variables either from the .env or from the os env variables are identical

**Comment by adimaralimuddin** [comment]
i'd solved it! in my case, i remove the semicolmn on at  .env file. probably my linter does that. but it's true. when i put any semicoln in the .env, it will show the unrelevant error.

**Reply by LeviMichalski** [reply]
This was my issue! IntelliJ used a semicolon to comment some code out, but then I added a .env extension and it used a # instead! I guess I never deleted the semicolon and once I did, it started working again.

**Comment by yahya-aghdam** [comment]
Just delete `'` or `"` from value of env when you entering this enviroment to Vercel :)

**Reply by manish-jsx** [reply]
after following your advice 

Checking if user exists in the database...
Login Error: PrismaClientKnownRequestError: 
Invalid `prisma.user.findUnique()` invocation:


Error validating datasource `db`: the URL must contain a valid API key

**Comment by noraa-july-stoke** [comment]
just fyi for sequelite it should be this: DATABASE_URL=file:./db.sqlite
works for me just fine.

**Comment by ullaskunder3** [comment]
I got this issue 
- Next.js 13 beta, supabase & prisma

```bash
[dxlord.🐼.007] final_portfolio (db_int)$ npx prisma db push
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database

Error: Prisma schema validation - (get-config wasm)
Error code: P1012
error: Environment variable not found: DATABASE_URL.
  -->  schema.prisma:10
   | 
 9 |   provider = "postgresql"
10 |   url      = env("DATABASE_URL")
   | 

Validation Error Count: 1
[Context: getConfig]

Prisma CLI Version : 4.13.0
```
## Then I just simply renamed the file `.env.local` to `.env` worked fine

```bash
[dxlord.🐼.007] final_portfolio (db_int)$ npx prisma db push
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "postgres", schema "public" at "localhost:54322"

🚀  Your database is now in sync with your Prisma schema. Done in 289ms

✔ Generated Prisma Client (4.13.0 | library) to ./node_modules/@prisma/
client in 356ms

```

**Reply by ullaskunder3** [reply]
Any tricks so we can use local and deploy.env

**Comment by Nicolius888** [comment]
just moving .env file to ./prisma/.env worked for me.

-Docker version 20.10.24, build 297e128
-Docker Compose version v2.18.1
-prisma/client: 4.12.0

**Comment by ArtemBorzienkov** [comment]
try to run `npx prisma generate` before the first connection to db, worked for me

**Comment by hoangnhat1997** [comment]
it can perfectly work. I work step-by-step as 
const prisma = new PrismaClient() in index.js after in the file
 schema.prisma  is     url      = env("DATABASE_URL")
 .env is DATABASE_URL="postgresql://postgres:123456@localhost:5432/postgres_dbname"

**Comment by Andrew-Sem** [comment]
I faced this issue when added prisma to existing project, using `.env.local`
`pnpx prisma init` created `.env` with sample `DATABASE_URL="file:./dev.db"`
So, just renamed my `.env.local` to `.env` and all fine

**Comment by davidawad** [comment]
nothing else here worked but the thing that did work for me was to use `?sslaccept=strict` instead of `ssl={"rejectUnauthorized":true}`

I'm working with prisma + trpc. 

I hope this helps someone 👌 

thanks everyone.

**Comment by ParasPandeyE** [comment]
Firstly thank you everyone,
I tried all the answers mentioned here but nothing worked for me, the issue with my project was that when I created the prisma.service.ts file and added url inside the constructor prettier added some extra inverted commas and saved the url as '"connection string"' and that was what was causing the error for me. Removing the extra inverted commas made the error go away.

Hope this helps if someone is also facing the same issue because it wasted a lot of time for me.

**Comment by TeaByte** [comment]
my problem was that
am using ( edge )

import { PrismaClient } from '@prisma/client/edge'
const prisma = new PrismaClient()

so i just used @prisma/client and it worked!
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()

**Reply by sylvinae** [reply]
am using prisma with planetscale and this fixed my thing! Thank youuuuu

**Reply by s1monhoffmann** [reply]
Fixed my Issue too, thanks

**Reply by YuriiSouza** [reply]
thank u man

**Reply by kishgau** [reply]
thanks mate. I was about to give up upon this client.

**Comment by martinslae** [comment]
I'm facing the same error and I did all the suggestions in this thread,  with no success.
My issue is on localHost in Windows.

I fixed!

The .env is loaded from OS environment variables first, and if is not there then is loaded from .env file.
In my case, I had an older "DATABASE_URL" variable in Windows, with whatever value, then my .env was ignored.
I fixed the value on the Windows variable and it works but will would ignore my configuration on .env, so I deleted the Windows variables so it works from my .env cause there's no key in the Windows environment anymore.

Don't forget to close and open the text editor (vscode) every time you change something in the Windows environment variable.

After wasting a lot of time I did it.  Thank god!

Thanks all

**Comment by imjotaefe** [comment]
Take attention when you're configuring your PrismaService. I was importing like this 
`import { PrismaClient } from '@prisma/client/edge';`

and I've changed for 
`import { PrismaClient } from '@prisma/client';`

This simple change worked for me. Hope help you!

**Reply by nikolasburk** [reply]
Thanks for sharing!

**Comment by Jeffr-K** [comment]
In my environment I resolved this issue

- MySQL / Prisma 2 / node.js(express) v16

   1. `git add` > `git commit -am "blah" ` > `git push heroku master`
   2. In Terminal: `heroku config:set DATABASE_URL="mysql://<your_prisma_url>" --app <your_heroku_app>`

**Comment by ordinz** [comment]
My issue is that my package.json I had `prisma generate --no-engine`.  Changing it to `prisma generate` resolved everything.

**Reply by favfly-dev** [reply]
this is the correct solution

**Comment by xyspg** [comment]
I solved it by changing `import { PrismaClient } from '@prisma/client/edge'`to `import { PrismaClient } from '@prisma/client'`, and remove `export const runtime = "edge";` so it uses Node.js runtime instead. Might be problem related to edge runtime

**Reply by jbcazaux** [reply]
i fixed it the same way.
For running next build on localhost with a localhost postgres, we cannot use the edge client in edge functions (seems fair since prisma cannot access our localhost)

**Reply by ytimocin** [reply]
This also solved my problem. Thanks @xyspg!

**Comment by temrb** [comment]
like others in the comment, I had tried prisma edge related products thus had

```
"build": "prisma generate --no-engine && next build",
```

in my scripts for build. Remove `--no-engine` as this is for edge.

**Comment by mudasirpandith** [comment]
I got this error while deploying to vercel. i need help

**Reply by safwansamsudeen** [reply]
Redeploying works for me. IDK why I have to do it everytime, though. You sure you running `generate` after `push`? You have to do that too.

**Comment by Abdullahijazvu** [comment]
I encountered an error saying that 

error: Error validating datasource `db`: the URL must start with the protocol `postgresql://` or `postgres://`.
  -->  schema.prisma:18
  
But after updating anything on schema.prisma or in env file
  
You must have to restart the application and prisma studio

Then error will be resolved

**Comment by martinslae** [comment]
My issue was I already has a key in the OS Environment Variable, so
instead get from my .env  it was getting from OS Variable, I deleted the OS
Variable and works fine.
The priority is OS then .env

Em ter., 16 de jan. de 2024 às 04:39, Abdullah Ijaz <
***@***.***> escreveu:

**Reply by ArianHamdi** [reply]
Wow, you saved me. thank you!

**Comment by icemagno** [comment]
Same in Blockscout backend .env file. Solved by removing double quotes. Use single quotes or no quotes at all.

**Comment by sashin123** [comment]
Issues occurred when there is an env.example, removing that file fixed my issues. No need to remove quotes. Also be sure the connection string has no errors after copying from db provider, i.e planetscale.

**Comment by jonaburg** [comment]
a very herculean effort to find that there was a whitespace at the end of the line ;)

**Reply by sashin123** [reply]
happens to the best of us 😂😂. it's the effort that counts right ?

**Comment by adamward459** [comment]
Just remote quotes in your connection, it will work

**Comment by George713** [comment]
I tried everything listed here. What solved it for me was the follwing:


```prisma
// schema.prisma
datasource db {
  provider  = "postgresql"
  url       = env("NEWNAME_DATABASE_URL")
}
```

```env
// .env
NEWNAME_DATABASE_URL=postgres://postgres....
```

After this I ran `npx prisma generate` to bring it to effect and started my app.

I conclude, that another environment variable named `DATABASE_URL` was present somewhere.

**Reply by johanguse** [reply]
Insane, this works for me!

**Reply by Dipak200** [reply]
I have done the above changes.
But now I am getting new error:
Error: 
Invalid `prisma.user.findFirst()` invocation:
The table `public.User` does not exist in the current database.
Source
src\lib\stripe.ts (24:17) @ async getUserSubscriptionPlan

  22 | }
  23 | 
> 24 | const dbUser = await db.user.findFirst({
     |               ^
  25 |   where: {
  26 |     id: user.id,
  27 |   },

Do you know how to resolve it?

**Reply by johanguse** [reply]
@Dipak200, please read the error message before asking! This is the error in your case: `The table public.User does not exist in the current database.` Did you run the command `prisma db push` ?

**Comment by matheusgondra** [comment]
I had the same problem. The solution was that there was already a DATABASE_URL variable defined in my OS. I had defined it for a java application. That's why prism gave an error, because the string started with "jdbc:" and not "postgresql://". Try checking in your terminal if the variable exists 

Linux
`echo $DATABASE_URL`

Windows
`echo %DATABASE_URL%`

**Comment by gijsroge** [comment]
Removing `node_modules` did the trick for me. Prisma probably caches some stuff.

**Comment by cesarlima-plank** [comment]
Change prisma's datasource provider from `sqlite` to `postgresql` or the provider that suits to your project.
```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

**Comment by CodyBontecou** [comment]
Fixed by deleting my `.nuxt` directory.

**Comment by iMeany** [comment]
Not Postgres, but for those who this might help, I ran into a very similar ambiguous error when trying to connect to [Azure SQL (MS SQL)](https://www.prisma.io/docs/orm/overview/databases/sql-server).

There seems to be some kind of bug/inconsistency with adding/not adding quotes (") around environment variables (shouldn't be needed) and then Prisma properly escaping them when reading `.env` combined with the [suggested {brackets} method](https://learn.microsoft.com/en-us/sql/connect/jdbc/building-the-connection-url?view=sql-server-ver16#escaping-values-in-the-connection-url) to escape special characters, so I recommend avoiding fields that need the escape (mainly the auto-generated password might be the issue).
schema.prisma:
```
datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
```
my `.env`
```
DATABASE_URL = "sqlserver://my-azure-db-here.database.windows.net:1433;database=my-database-name;user=myadminuser@database-server-name;password=stillSafePasswordWithoutNeedingTheEscape123123;encrypt=true;trustServerCertificate=false;hostNameInCertificate=*.database.windows.net;loginTimeout=30;"
``` 

You can still have special characters in the password, but by avoiding the ones that need escaping you can copy the connection string from Azure DB connection panel, remove the "jdbc:" from the start and it works. I tried with/no quotes, with/no brackets and I couldn't get it working by trying all the combinations.

**Comment by SirawichDev** [comment]
I used .env.local it not working. But the error was gone when I changed from .env.local to .env file. seem  ``` env("DATABASE_URL") ``` read only .env file

**Reply by imyuanli** [reply]
Yeah I can't get variables from .env.local either, waiting for a solution.

**Reply by SirawichDev** [reply]
i think to workaround we need to declare a key/value into .env instead.
or we have to use  `dotenv`  and put this script into package.json or run in the command line likes 
`dotenv -e .env.local -- npx prisma generate`

**Comment by AkramMahmoudi** [comment]
### **my solution for this problem is make new env variable like DB or DBLINK and i work for me**

**Reply by vikas-ukani** [reply]
I've used different variable names but not worked for me.

**Comment by eltonsampaio12** [comment]
In my case, I was using env.local files, but at some point, a .env file was generated unexpectedly. I was referencing env.local, but I should have been looking at .env. To resolve this, you can either rename your env.local file to .env or move the variables from env.local into the .env file directly.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by bruno-pmakers** [comment]
For me, I incorrectly added `--no-engine` to generate command while using a non Prisma Postgres database.
Before:
`"db:generate": "prisma generate --no-engine"`

After
`"db:generate": "prisma generate"`

The docs says:

> If you are not using a Prisma Postgres database, remove the --no-engine flag from the custom scripts above.

**Reply by B3nsten** [reply]
Reading docs is only for the weak, I guess. Thanks for pointing it out.

**Reply by JaeHye0k** [reply]
Thanks! I worked for me.

**Reply by aromibenard** [reply]
This worked for me.

**Reply by Roman86** [reply]
in my case I forgot to remove `--no-engine` after disabling the Prisma Accelerate (which required that flag). The error is gone

**Comment by welikeheon** [comment]
I found a solution.
In my case, env was wrong.

I did used mixed quotes **'** and **"**
```env
DATABASE_URL="mysql://id:pw@mariadb:3306/work'

// Use pair with "
DATABASE_URL="mysql://id:pw@mariadb:3306/work"

and it works fine.

**Comment by Amir-A-M** [comment]
In my case, I had created a new database in a different server. 
### I forgot to run the blow commands:
```
npx prisma migrate dev --name init --create-only 
npx prisma generate
```

**Comment by m-elagamy** [comment]
The problem was solved by just adding it to repo secrets so the build can access them: 👇

<img width="778" height="185" alt="image" src="https://github.com/user-attachments/assets/0b33dc48-96fa-4542-9228-664d02414d12" />

**Comment by imyuanli** [comment]
这是自动回复的邮件。您好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。
